Add Lesson 7 notebook

canyon289 opened the pull request:
Main notebook is work in progress. Need help with a particular issue hence push

canyon289 commented:
@tomicapretto @AlexAndorra ready for code, model, and high level narration review. If you like it I'll fill it in with more details and narration

tomicapretto commented:
**Summary:** what we have in the notebook is good and is on the right track. I wouldn't remove it (or I would keep almost all of it). That said, I think it still needs some work (see the comments)

---

**More and more custom visualizations**

* In practice visualizations “just” need to be able to share a message
* But we all love visualizations that look nice, and I think students value it a lot
* For example “Total Goals Per team” can be a visualization. I imagine a barplot with numbers on top of the bars.
* Don't forget appropriate labels

**Description of the variables**

* I see a bunch of names, what do they mean? Which ones are the important?
* Give more context about the data, where did you take it from?
* How many observations? How many variables? Are them of the right type?

**Simulated data before real data**
* I think it would be good students leave the section with two practical examples (as it happens in Categorical regression for example)
* The first one would be the introduction, very easy example, where we explain the gist of it and extract main conclusions
* I imagine a single section where we cover it and then we conclude along the lines “This is all about Poisson regression, see how trivial it is... well, it's not." and then move onto the real life example and we work it out more in detail and we actually face modeling problems
    * The second example would be the real data

**Use aliases**
* Some term names end up being too long. We can show-off aliases in Bambi.

**Zero-Inflated-Poisson**
* Ideally, it would be great to have a real example where there's zero inflation. I'll try to look for one.


This may be a matter of taste, but I find code like this hard to understand since there's A LOT going on

```python
(long_df.query("Team in ('Man City','Huddersfield') ")[["Team","Goals"]]
            .groupby(["Goals", "Team"]).size()
            .reset_index().pivot(index="Goals", columns="Team")
            .droplevel(0, axis='columns').plot(kind="bar"))
```
I would split it into different “lines” or “chunks”

canyon289 commented:
Notebook definitely need a lot of work (thats on me)

For simulated data do you mean we start with a simulated dataset and go all the way through the first sections, and then the last section is the football example. Similar to what we do in Link Functions and Categorical.

Do you mean in each section we simulate data first before then football regression? 

> Ideally, it would be great to have a real example where there's zero inflation. I'll try to look for one.

Thanks, I was struggling on finding a good one


> This may be a matter of taste, but I find code like this hard to understand since there's A LOT going on
Thats funny because I originally had it as separate lines, then I thought youd tell me make it more "modern" as a one liner so I figured out how to that :sweat: 

Ill change it back no worries!

canyon289 commented:
Section 10: Simulated poisson no covariates, 1 covariate
  * This poisson poisson link
  * This is poisson and lambvda and natural and paramat r+
Section 20: EDA  and no covariate
Section 30: Home field advantage
Section 40 : perform,ance over time
Section 50: Zip but with new dataset
  * Another team inflate the zeros


Ravin to provide the fit with all teams, see if there is a team with zero inflation

canyon289 commented:
I addressed all comments above except
### Using non simulated data for ZIP
I didn't find any teams that had a big zero inflation. Simulating seems still good, I think I can make this work if I change the team name and explain it well. What do you both think

### Modeling marginals
all teams together still looks poisson. If the idea is to explain the marginal concept though maybe we should do it in another lesson, like lesson 4 as that concept has already been happening implicitly.

## Variable names from football example
Most of them dont matter, I explained the ones that do more in depth

With that most of the notebook is done except Section 60 for ZIP. Once we get agreement I can finish that off too.

Look over the notebook and let me know what you both think!

canyon289 commented:
Ah and in regards to plot sizes i'll fix that when I record because when I record on my home desktop i can get the sized dialed in correctly. Feel free to leave any comments!

tomicapretto commented on lessons/06_CountModels/TeamPerformanceOverTime.ipynb:
```diff
@@ -0,0 +1,1719 @@
+{
```
<p>test</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

tomicapretto replied:
<p>asdf</p>

tomicapretto commented on lessons/06_CountModels/TeamPerformanceOverTime.ipynb:
```diff
@@ -0,0 +1,1719 @@
+{
```
<p>test</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

canyon289 commented:
@tomicapretto every suggestion is fantastic. Will implement them all

tomicapretto commented:
<p>Also make sure it has appropriate labels </p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746955'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>I'm not sure how to make this visualiation. I added a temporary one but i dont think its one you intended</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746800'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Renamed the variables. Leaving open for confirmation. Ill run black on the code before publishing</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746817'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>There was a datetime parsing error. Thanks for catching that</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746880'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p> </p><p>Added the model mathematically. Leaving open until its clear you like it</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488621079'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
Two ideas
* One is good version of matrix approach
* Lesson 3, all equations with different coefficients, more parameters than things you want to estimate
  * Perhaps come up with an equivalent version of that. There

tomicapretto commented:
![image](https://user-images.githubusercontent.com/25507629/229940440-0aab50d9-19e3-42b1-bd7e-9cd882df3821.png)

canyon289 commented:
Ready for review again

Made more updates
1 . A few comments are left open from the last review
2. Biggest discussion is whether zip should stay in this or move to the next lesson
3. Aliases are an open item. Theyre part of section 60 though so if we move them its not a concern here
4. In section 50 I'm getting an exception with xarray group by that maybe theres an obvious solution to. If not I'll delete it and keep the more verbose pandas work around

tomicapretto commented:
<p>Wonderful</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746817'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tomicapretto commented:
<p>premeire -&gt; premier</p><p> </p><p>And you can say 2017-2018 season</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746865'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tomicapretto commented:
<p>I refer to the fact the uncertainty in the posteriror predictive distribution is quite large (the overall shape resembles the observed data well, but the "band" is quite thick, meaning individual draws of the posterior predictive distribution are very different)</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488620952'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tomicapretto commented:
<p>Review the sentence "In categorical each outcome is independent"</p><p> </p><p>Then it says "its uniform" -&gt; "it's uniform"</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488621055'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tomicapretto commented:
<p>Looks amazing. The only nit is lambda needs the "i" subscript (because it comes from the nu_i)</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488621079'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Sorry I didnt ocmment on this earlier. I'll have Fadi animate them in</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746906'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

tomicapretto commented:
<p>love it</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746906'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202142'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202154'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Done</p><p> </p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202163'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202172'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202186'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Added</p><p> </p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202203'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>No worries on the typos, I'll fix them all in the final pass during recording.</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202212'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202225'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>done, curious why var? because its a bigger difference?</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202236'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202265'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202273'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202288'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202351'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1487746865'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202360'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202369'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488620882'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>Youre right. Fixed to </p><p> </p><p>All we need to do is use our new likelihood and link function!</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202381'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488620952'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>removed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202396'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1488621079'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>fixed</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202411'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

canyon289 commented:
<p>done</p>

---

 <a href='https://app.reviewnb.com/tomicapretto/etc/pull/26/discussion/#comment-1504202422'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>
